### ISSUE TITLE
class-string instantiation does not flag consumed variable as being used

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#1384]
# issue
When instantiating an object via a constructor specified with `class-string`, the consumed argument does not get flagged as used.
Practical example is when assigning the results of `PDOStatement::fetchAll()` to `$data`, then consuming `$data[0]` in a constructor.
Minimal example provided with `ArrayObject` below.

# repro
* https://psalm.dev/r/443197c0ee
* using 674657a

```php
<?php
/**
* @return array[]
*/
function foo() : array
{
  return [['foo'], ['bar']];
}

/**
* @param class-string<ArrayObject> $type
*/
function bar(string $type) : ? ArrayObject {
  /**
  * @var array[]
  */
  $data = foo();
  
  if (random_int(0, 1) === 1) {
  	return new $type($data[0]);
  }
  
  return null;
}

function baz() : ? ArrayObject {
  /**
  * @var array[]
  */
  $data = foo();
  
  if (random_int(0, 1) === 1) {
  	return new ArrayObject($data[0]);
  }
  
  return null;
}
```

> Psalm output (using commit 674657a): 
> 
> INFO: UnusedVariable - 17:3 - Variable $data is never referenced

**Comment by muglug** [comment]
oof, that's because it's not even analysed. Thanks!
